modules/nats/nats.go

```diff
@@ -58,6 +58,19 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 		genericContainerReq.Cmd = append(genericContainerReq.Cmd, []string{"--" + k, v}...)
 	}
 
+	// Pass a configuration file if provided
+	if settings.ConfigFile != nil {
+		genericContainerReq.Cmd = append(genericContainerReq.Cmd, "-config", "/etc/nats.conf")
+		genericContainerReq.Files = append(
+			genericContainerReq.Files,
+			testcontainers.ContainerFile{
+				Reader:            settings.ConfigFile,
+				ContainerFilePath: "/etc/nats.conf",
+				FileMode:          0o644,
+			},
+		)
+	}
```

Comment:
As @stevenh mentioned, this block is the one that should go into the functional option.

Other than that, thanks for adding this, looks promising! 🙏

Reply:
https://github.com/testcontainers/testcontainers-go/pull/2905/commits/5da61731f5dfb3f77ab224f22ba36e3149307042